homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return SUPPORT_ON_OFF | SUPPORT_STREAM
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.device.online
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        if self.enabled:
+            return "mdi:camcorder"
+        return "mdi:camcorder-off"
+
+    @property
+    def state(self):
+        """Return the camera state."""
+        if self.device.recording:
+            return STATE_RECORDING
+        if self.device.online:
+            return STATE_STREAMING
+        return STATE_IDLE
+
+    async def stream_source(self) -> str:
+        """Return the mp4 stream source."""
+        return self._stream_url
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this agent object."""
+        return f"{self._unique_id}"
+
+    async def async_enable_alerts(self):
+        """Enable alerts."""
+        await self.device.alerts_on()
+
+    async def async_disable_alerts(self):
+        """Disable alerts."""
+        await self.device.alerts_off()
+
+    async def async_enable_motion_detection(self):
+        """Enable motion detection."""
+        await self.device.detector_on()
+
+    async def async_disable_motion_detection(self):
+        """Disable motion detection."""
+        await self.device.detector_off()
+
+    async def async_start_recording(self):
+        """Start recording."""
+        await self.device.record()
+
+    async def async_stop_recording(self):
+        """Stop recording."""
+        await self.device.record_stop()
+
+    async def async_turn_on(self):
+        """Enable the camera."""
+        await self.device.enable()
+
+    async def async_toggle(self):
+        """Enable/disable the camera."""
+        if self.device.online:
+            await self.device.disable()
+        else:
+            await self.device.enable()
+
+    async def async_snapshot(self):
+        """Take a snapshot."""
+        await self.device.snapshot()
+
+    async def async_turn_off(self):
+        """Disable the camera."""
+        await self.device.disable()
+
+    # Other Entity method overrides
+
+    async def async_added_to_hass(self):
+        """Subscribe to signals and add camera to list."""
+        for service, method in CAMERA_SERVICES.items():
+            self.async_on_remove(
+                async_dispatcher_connect(
+                    self.hass,
+                    service_signal(service, self.entity_id),
+                    getattr(self, method),
+                )
+            )
```

Comment:
I'm pretty sure you can delete this as `async_register_entity_service` will call `service.entity_service_call` which will call `_handle_entity_call` which will end up calling `getattr(entity, func)`